msg/EntityInfo.msg

```diff
@@ -0,0 +1,14 @@
+# Entity type and additional information
+
+uint8 OBJECT         = 0    # Generic or unspecified type.
+uint8 ROBOT          = 1    # A broad category for mobile robots, arms, drones etc., usually with ROS 2 interfaces.
+uint8 HUMAN          = 2    # Simulated humans, e.g. pedestrians, warehouse workers. Compared to DYNAMIC_OBJECT category,
+                            # humans are often expected to be treated exceptionally in regards to safety constraints.
+uint8 DYNAMIC_OBJECT = 4    # Vehicles, animals, mobile obstacles, typically to present a detection & tracking challenge,
+                            # not themselves subject to validation.
+uint8 STATIC_OBJECT  = 5    # Any object which is static, e.g. not supposed to change its pose
+                            # unless by means of SetEntityState.
+
+uint8 category              # Major category for the entity. Extra entity type distinction can be made through tags.
```

Comment:
Not quite what I had in mind re: categories.

In my opinion, enumerating all possible types of entity categories is impossible, so I was just saying to add a simple string so that developers/users could add their own domain-specific categories.

That being said, your categories are pretty broad so this is probably fine. But I think there also needs to be a short-form user category (e.g., "furniture", "food") as well as a long-form description as you have.

Reply:
Looking at it now, I'd suggest moving definitions of the constants to a separate message file. This has the bonus that it is super easy to add more in the future without changing the hashes and binary representation of the EntityInfo message (as it only defines integer "aliases" and EntityInfo uses just int type).

Reply:
Call me crazy but it feels like categories and tags are a duplicate. What's the benefit of having 2 separate fields? Couldn't categories simply be pre-defined tags ?